tests/edalize_common.py

```diff
@@ -1,21 +1,90 @@
-import os.path
-import tempfile
 from collections import OrderedDict
+import filecmp
+import os.path
+import shutil
+
+import pytest
 
 from edalize import get_edatool
 
+
 tests_dir = os.path.dirname(__file__)
 
+
+class TestFixture:
+    def __init__(self, test_name, tool_name, param_types, files,
+                 tool_options, ref_dir, work_root, use_vpi):
+        self.test_name = test_name
+        self.ref_dir = ref_dir
+        self.work_root = work_root
+        self.backend = _setup_backend(test_name, tool_name, param_types, files,
+                                      tool_options, work_root, use_vpi)
+
+    def compare_files(self, files, ref_subdir='.'):
+        ref_dir = os.path.normpath(os.path.join(self.ref_dir, ref_subdir))
+        return compare_files(ref_dir, self.work_root, files)
+
+    def copy_to_work_root(self, path):
+        shutil.copy(os.path.join(self.ref_dir, path),
+                    os.path.join(self.work_root, path))
+
+
+@pytest.fixture
+def make_edalize_test(monkeypatch, tmpdir):
+    '''A factory fixture to make an edalize backend with work_root directory
+
+    The returned factory method takes a tool_name (the name of the tool) and
+    the following keyword arguments.
+
+    test_name: The name to call the backend. Default: test_<tool_name>_0
+
+    param_types: A list of parameter types.
+                 Default: ['plusarg', 'vlogdefine', 'vlogparam'] (the parameter
+                 types supported by most simulators).
+
+    files: A list of files to use. If None (the default) use FILES.
+
+    tool_options: Dictionary passed to _setup_backend. Default: {}.
+
+    ref_dir: A reference directory relative to test_<tool_name>. Default: '.'
+
+    use_vpi: If true, set up backend with definitions from :attr:`VPI`.
+
+    '''
+    # Prepend directory `mock_commands` to PATH environment variable
+    monkeypatch.setenv('PATH', os.path.join(tests_dir, 'mock_commands'), ':')
+
+    created = []
+
+    def _fun(tool_name,
+             test_name=None,
+             param_types=['plusarg', 'vlogdefine', 'vlogparam'],
+             files=None,
+             tool_options={},
+             ref_dir='.',
+             use_vpi=False):
+        if test_name is None:
+            test_name = 'test_{}_0'.format(tool_name)
+        work_root = tmpdir / str(len(created))
+        work_root.mkdir()
+        ref_dir = os.path.normpath(os.path.join(tests_dir,
+                                                'test_' + tool_name,
+                                                ref_dir))
+        fixture = TestFixture(test_name, tool_name, param_types, files,
+                              tool_options, ref_dir, str(work_root), use_vpi)
+        created.append(fixture)
+        return fixture
+
+    yield _fun
```

Comment:
What's the benefit of using a `yield` fixture here? There's no tear-down code after the yield, so you can make this a simple `return`.